nucypher/blockchain/eth/registry.py

```diff
@@ -440,3 +449,32 @@ def read(self) -> list:
             else:
                 raise
         return registry_data
+
+
+class IndividualAllocationRegistry(InMemoryAllocationRegistry):
+
+    REGISTRY_NAME = "individual_allocation_ABI.json"
+
+    def __init__(self, beneficiary_address: str, contract_address: str, contract_abi=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.beneficiary_address = beneficiary_address
+        self.contract_address = contract_address
+
+        if not contract_abi:
+            # Download individual allocation template to extract contract_abi
+            individual_allocation_template = json.loads(self.fetch_latest_publication())
+            contract_data = [*individual_allocation_template.values()].pop()
```

Comment:
Ugh, what's happening here?
Upon closer inspection,
```
inidividual_allocation_template = {'BENEFICIARY_ADDRESS':
    ['ALLOCATION_CONTRACT_ADDRESS', <abi>}
```
Can `BENEFICIARY_ADDRESS` be an address, or is it always a template if we are here, and can be accessed by key?

Reply:
Main concern with the method above is that the dict returned by json.load is not ordered (although in python 3.7 it is). But it has only one value...

Reply:
Yes, the template is always of the form you wrote above, with just one entry. I added a check that raises an exception otherwise.